Bio/SeqIO/InsdcIO.py

```diff
@@ -1117,6 +1118,17 @@ def write_record(self, record):
         handle.write("FEATURES             Location/Qualifiers\n")
         rec_length = len(record)
         for feature in record.features:
+            # Handle any stray line breaks in the feature
+            for k in feature.qualifiers.keys():
+                if (
+                    isinstance(feature.qualifiers[k], Iterable)
+                    and "\n" in feature.qualifiers[k]
+                ):
+                    feature.qualifiers[k] = feature.qualifiers[k].replace("\n", " ")
+                    warnings.warn(
```

Comment:
Warnings are not really intended for this kind of use as they are issued only once in a Python session. So if you parse two files that have a stray line break, the first one will trigger the warning but the second one will not.

Reply:
would it be better to have a variable in a higher scope that keeps track of the number of times a line break is stitched and report it after the file is done being written?

Reply:
I would prefer some argument to `write_record` that indicates whether the parser should raise a `ValueError` if a stray line break is found, or otherwise just replace the stray line error without raising a warning. For example, the `parse` function in `Bio.motifs` has a `strict` argument. If `True` (the default), the parser will raise a `ValueError` if the file does not strictly comply with the file format.

As an example of how treacherous warnings are, try the following:
```python
>>> import warnings
>>> def f():
...     warnings.warn("consider this your last warning!")
...
>>> def g():
...     warnings.warn("consider this your last warning!")
...
>>> f()
<stdin>:2: UserWarning: consider this your last warning!
>>> g()
>>> # no warning
```

Reply:
I'm not getting the same results as you in your script:

```python
>>> import warnings
>>> def f():
...  warnings.warn("warn 1")
...
>>> def g():
...  warnings.warn("warn 2")
...
>>> f()
<stdin>:2: UserWarning: warn 1
>>> g()
<stdin>:2: UserWarning: warn 2
>>>
```

Furthermore:
```python
>>> for i in range(10):
...  warnings.warn(f"warn{i}")
...
<stdin>:2: UserWarning: warn0
<stdin>:2: UserWarning: warn1
<stdin>:2: UserWarning: warn2
<stdin>:2: UserWarning: warn3
<stdin>:2: UserWarning: warn4
<stdin>:2: UserWarning: warn5
<stdin>:2: UserWarning: warn6
<stdin>:2: UserWarning: warn7
<stdin>:2: UserWarning: warn8
<stdin>:2: UserWarning: warn9
>>>
```

I'm using CPython and 3.9, maybe it's different in my version?

Reply:
Including the line in the message would ensure it gets seen multiple times with the default warning settings.

Adding a strict mode is beyond the scope of this issue, but could make sense. We have a precedent for this with the PDB parser. Personally I prefer the warnings approach - the user can use the warnings module to halt on a specific warning.

Reply:
@peterjc 
> Including the line in the message would ensure it gets seen multiple times with the default warning settings.

It won't ensure that. For example, if a user tries to parse the same file twice, the warning will not appear the second time.

Reply:
Pathological, but yes. I was thinking of multiple "funny" feature annotations in a single file.

The default warnings filters try to balance verbosity vs information.